scanpy/readwrite.py

```diff
@@ -277,15 +277,49 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = dict(obs_names=dsets['barcodes'].astype(str))
+            var_dict = dict(
+                var_names=dsets['name'].astype(str),
+            )
+            if 'gene_id' not in dsets:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['id'].astype(str),
+                        'feature_types': dsets['feature_type'].astype(str),
+                    }
+                )
+            else:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['gene_id'].astype(str),
```

Comment:
The `gene_id` is not unique in a probe-barcode matrix. Is scanpy okay with that?

Reply:
`gene_id` should be fine, but their `var` is a dataframe generated using `var_dict` with `var_name` as the index. For feature barcode matrices, this uses `name` as the index - which is gene names and has repetitions (but they must have their reasons for that design.).